### DISCUSSION TITLE
Presolving in Dual Task

### DISCUSSION THREAD

**Discussion by yusiy** [ds4dm/ml4co-competition#12] [Q&A]
Hi,
We are wondering if ecole will presolve the problems by default in dual task (load balancing problem), or adding presolving method is allowed. 
Not sure whether this behavior will be regarded as an interaction with the solver.
Best

**Comment by gasse** [comment]
Hi @yusiy ,

Yes, MILP instances will have been presolved in the dual task, with SCIP's default parameters. The only change in this environment, with respect to vanilla SCIP, is that we deactivate primal heuristics and restarts (see the dual environment dynamics [here](https://github.com/ds4dm/ml4co-competition/blob/main/common/environments.py#L76)).

You cannot alter or add presolving methods in the dual task, you are only allowed to perform branching.

Best,
Maxime
